docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+## State
+
+Props are set by the parent component and can only be changed by the parent component passing down new values. But for individual components to change according to human interactions, they need to be able to accept and store their own data. Fortunately, React components can have their own `state` you can get and set data from.
+
+<div class="toggler">
+  <ul role="tablist" class="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTabs('syntax', 'functional')">
+      State with Function Components
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTabs('syntax', 'classical')">
+      State with Class Components
+    </li>
+  </ul>
+</div>
+
+<block class="functional syntax" />
+
+You can add state to a component by calling [React’s `useState` Hook](https://reactjs.org/docs/hooks-state.html).
+
+> A Hook is a kind of function that lets you “hook into” React features. For example, `useState` is a Hook that lets you add state to function components. You can learn more about [other kinds of Hooks in the React documentation.](https://reactjs.org/docs/hooks-intro.html)
+
+You can declare a component’s state by calling `useState` inside its function like this:
+
+`const [currentValue, setNewValue] = useState("Initial value.");`
+
+You can use `useState` to track any kind of data, including numbers:
+
+`const [quantity, pickQuantity] = useState(0);`
+
+Calling `useState` does two things:
+
+- it creates a “state variable” with an initial value
+- it creates a function to set that state variable’s value
+
+Let’s break `[a, b] = useState(c)` down:
+
+- `a` is the “state variable”—it refers to the current value of the component’s state
+- `b` is the name of the function you use to update the state variable
+- `c` is the value you initialize the state variable with
+
+```SnackPlayer name=State
+import React, { useState } from 'react';
+import { Text, TextInput, View } from 'react-native';
+
+function Cat() {
+
+  const [name, setName] = useState("What's my name again?");
+
+  return (
+    <View>
+      <Text>Hello, I am...{name}!</Text>
+      <TextInput
+        style={{
+          height: 40,
+          borderColor: 'gray',
+          borderWidth: 1
+        }}
+        onChangeText={
+          (text) => setName(text)
+        }
+      />
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <>
+      <Cat />
+      <Cat />
+    </>
+  );
+}
+```
+
+> See the `<>` and `</>` above? These bits of JSX are [fragments](https://reactjs.org/docs/fragments.html). Adjacent JSX elements must be wrapped in an enclosing tag. Fragments let you do that without nesting an extra, unnecessary wrapping element like `View`.
+
+First, you will want to import `useState` from React like so:
```

Comment:
I feel like this section repeats the previous one. Maybe you can axe the previous one and just do this explanation.

Reply:
Or maybe they could be swapped around?